engine/server/engine/centralized_logs/client_implementations/persistent_volume/stream_logs_strategy/per_week_stream_logs_strategy.go

```diff
@@ -380,6 +387,11 @@ func (strategy *PerWeekStreamLogsStrategy) followLogs(
 	return nil
 }
 
+func isValidJsonString(line string) bool {
+	var jsonLog JsonLog
+	return json.Unmarshal([]byte(line), &jsonLog) == nil
+}
```

Comment:
Is it going to be expensive to do this test on every line that we process (thinking about the memory and cpu resources that are spent)?

Reply:
Hm good point - we have to do this operation anyways to process the message (extract the log message field from the log line). I could refactor so that this call is only being made once.

Reply:
If we find that making this call even once per log line is inefficient, in the future can optimize to unmarshal several json log lines at once.